__doctest_requires__ doesn't recognize subpackages any more

The followings used to work:

```
astropy/coordinates/distances.py:__doctest_requires__ = {'*': ['scipy.integrate']}
astropy/cosmology/core.py:__doctest_requires__ = {'*': ['scipy.integrate', 'scipy.special']}
astropy/cosmology/funcs.py:__doctest_requires__ = {'*': ['scipy.integrate']}
astropy/stats/funcs.py:__doctest_requires__ = {'binom_conf_interval': ['scipy.special'],
astropy/stats/circstats.py:__doctest_requires__ = {'vtest': ['scipy.stats']}
astropy/stats/jackknife.py:__doctest_requires__ = {'jackknife_stats': ['scipy.special']}
```

However with the 0.4.0 release errors are raised for all these cases when scipy is not available:
```
______________ ERROR collecting astropy/coordinates/distances.py _______________
../../../miniconda/envs/test/lib/python3.6/site-packages/pytest_doctestplus/plugin.py:137: in collect
    for test in finder.find(module):
../../../miniconda/envs/test/lib/python3.6/site-packages/pytest_doctestplus/plugin.py:456: in find
    tests = list(filter(test_filter, tests))
../../../miniconda/envs/test/lib/python3.6/site-packages/pytest_doctestplus/plugin.py:452: in test_filter
    if not self.check_required_modules(mods):
../../../miniconda/envs/test/lib/python3.6/site-packages/pytest_doctestplus/plugin.py:413: in check_required_modules
    module = importlib.util.find_spec(mod)
../../../miniconda/envs/test/lib/python3.6/importlib/util.py:88: in find_spec
    parent = __import__(parent_name, fromlist=['__path__'])
E   ModuleNotFoundError: No module named 'scipy'
```
https://travis-ci.org/astropy/astropy/jobs/586739534#L1279

--- a/pytest_doctestplus/plugin.py
+++ b/pytest_doctestplus/plugin.py
@@ -7,7 +7,7 @@ import six
 
 import doctest
 import fnmatch
-import importlib
+import importlib.util
 import os
 import re
 import sys
@@ -410,7 +410,10 @@ class DocTestFinderPlus(doctest.DocTestFinder):
                 except ImportError:
                     module = None
             else:
-                module = importlib.util.find_spec(mod)
+                try:
+                    module = importlib.util.find_spec(mod)
+                except ModuleNotFoundError:
+                    module = None
 
             if module is None:
                 cls._import_cache[mod] = False

--- a/pytest_doctestplus/plugin.py
+++ b/pytest_doctestplus/plugin.py
@@ -412,7 +412,7 @@ class DocTestFinderPlus(doctest.DocTestFinder):
                 import importlib.util
                 try:
                     module = importlib.util.find_spec(mod)
-                except ModuleNotFoundError:
+                except ImportError:
                     module = None
 
             if module is None:

--- a/pytest_doctestplus/plugin.py
+++ b/pytest_doctestplus/plugin.py
@@ -7,7 +7,6 @@ import six
 
 import doctest
 import fnmatch
-import importlib.util
 import os
 import re
 import sys
@@ -410,6 +409,7 @@ class DocTestFinderPlus(doctest.DocTestFinder):
                 except ImportError:
                     module = None
             else:
+                import importlib.util
                 try:
                     module = importlib.util.find_spec(mod)
                 except ModuleNotFoundError:

--- a/tests/python/doctests.py
+++ b/tests/python/doctests.py
@@ -8,6 +8,7 @@ __doctest_skip__ = [
 
 __doctest_requires__ = {
     'depends_on_foobar': ['foobar'],
+    'depends_on_foobar_submodule': ['foobar.baz'],
     'depends_on_two_modules': ['os', 'foobar'],
 }
 
@@ -50,6 +51,16 @@ def depends_on_foobar():
     """
 
 
+def depends_on_foobar_submodule():
+    """
+    This test will cause a failure if __doctest_requires__ is not working.
+
+    >>> import foobar.baz
+    >>> foobar.baz.bar('baz')
+    42
+    """
+
+
 def depends_on_two_modules():
     """
     This test will cause a failure if __doctest_requires__ is not working.
@@ -80,7 +91,7 @@ class ClassWithSomeBadDocTests(object):
 
 
 class ClassWithAllBadDocTests(object):
-    def this_test_fails():
+    def this_test_fails(self):
         """
         This test will cause a failure if __doctest_skip__ is not working.
 
@@ -88,7 +99,7 @@ class ClassWithAllBadDocTests(object):
         5
         """
 
-    def this_test_also_fails():
+    def this_test_also_fails(self):
         """
         This test will cause a failure if __doctest_skip__ is not working.
 

